### PR TITLE
fix(#2406): added test for FakeMaven

### PR DESCRIPTION
**Pull request by Yanich96**
Ref: #2406

What's done:

1.   I have found the bug: in FakeMaven parameters were not configured for AssembleMojo.
2.  I have added test, that checks this bug.
3.  I have added todo for fixing this bug.

---

## PR-Codex overview
This PR focuses on fixing the configuration of custom parameters in the `FakeMaven` class. 

### Detailed summary
- Fixed the `FakeMaven#allowedParams(Class)` function to configure parameters of executed Mojo and parameters of Mojos inside the executed Mojo.
- Added a test case to ensure that custom parameters can be configured within the `AssembleMojo`.

> ✨ Ask PR-Codex anything about this PR by commenting with `/codex {your question}`

### CODE REVIEW

**Comment by maxonfjvipon** [comment]
@Yanich96 please add:
- reference to main issue
- explanation what kind of bug is found
- reason for removing `trackOptimizationSteps` from `AssembleMojo`

File: eo-maven-plugin/src/test/java/org/eolang/maven/AssembleMojoTest.java [hunk 1]
```diff
@@ -261,6 +261,28 @@ void assemblesSuccessfullyInOfflineMode(final Logs out, @TempDir final Path temp
...
+    @Test
+    @Disabled
+    void configuresChildParameters(@TempDir final Path temp) throws IOException {
+        final Map<String, Path> res = new FakeMaven(temp)
+            .withHelloWorld()
+            .with("trackOptimizationSteps", true)
+            .execute(AssembleMojo.class)
+            .result();
+        MatcherAssert.assertThat(
+            res,
```
Comment (maxonfjvipon) [inline eo-maven-plugin/src/test/java/org/eolang/maven/AssembleMojoTest.java:273]: @Yanich96 let's add assertion message here and below

File: eo-maven-plugin/src/test/java/org/eolang/maven/FakeMaven.java [hunk 2]
```diff
@@ -70,6 +70,13 @@
  * their behaviour and results.
  * NOT thread-safe.
  * @since 0.28.12
+ * @todo #2406:30min Fix {@link FakeMaven#allowedParams(Class)}
+ *  This function parameters of executed Mojo in {@link FakeMaven#execute(Class)}
```
Comment (maxonfjvipon) [inline eo-maven-plugin/src/test/java/org/eolang/maven/FakeMaven.java:74]: @Yanich96 I think English needs to be corrected here. Please use translator or chatGPT

**Review by maxonfjvipon — CHANGES_REQUESTED** [review]
@Yanich96 good one, just a few comments from my side

File: eo-maven-plugin/src/test/java/org/eolang/maven/AssembleMojoTest.java (hunk 1 above)
Comment (Yanich96) [reply eo-maven-plugin/src/test/java/org/eolang/maven/AssembleMojoTest.java:274]: I have done

File: eo-maven-plugin/src/test/java/org/eolang/maven/FakeMaven.java (hunk 2 above)
Comment (Yanich96) [reply eo-maven-plugin/src/test/java/org/eolang/maven/FakeMaven.java]: I have changed

**Review by maxonfjvipon — APPROVED** [review]
@Yanich96 LGTM! Thanks

**Comment by Yanich96** [comment]
@yegor256 please approve

**Pull request merged**